src/storage-preview/azext_storage_preview/operations/account.py

```diff
@@ -143,6 +143,60 @@ def update_storage_account(cmd, instance, sku=None, tags=None, custom_domain=Non
     return params
 
 
+def create_blob_inventory_policy(cmd, client, resource_group_name, account_name, policy):
+    # BlobInventoryPolicy = cmd.get_models('BlobInventoryPolicy')
+    # TODO: add again with rule management if bandwidth is allowed
+    # BlobInventoryPolicy, BlobInventoryPolicySchema, BlobInventoryPolicyRule, BlobInventoryPolicyDefinition, \
+    # BlobInventoryPolicyFilter = cmd.get_models('BlobInventoryPolicy', 'BlobInventoryPolicySchema',
+    #                                            'BlobInventoryPolicyRule', 'BlobInventoryPolicyDefinition',
+    #                                            'BlobInventoryPolicyFilter')
+    # filters = BlobInventoryPolicyFilter(prefix_match=prefix_match, blob_types=blob_types,
+    #                                     include_blob_versions=include_blob_versions,
+    #                                     include_snapshots=include_snapshots)
+    # rule = BlobInventoryPolicyRule(enabled=True, name=rule_name,
+    #                                definition=BlobInventoryPolicyDefinition(filters=filters))
+    # policy = BlobInventoryPolicySchema(enabled=enabled, destination=destination,
+    #                                    type=type, rules=[rule])
+    # blob_inventory_policy = BlobInventoryPolicy(policy=policy)
+    #
+    # return client.create_or_update(resource_group_name=resource_group_name, account_name=account_name,
+    #                                blob_inventory_policy_name=blob_inventory_policy_name,
+    #                                properties=blob_inventory_policy,
+    #                                **kwargs)
+    if os.path.exists(policy):
+        policy = get_file_json(policy)
+    else:
+        policy = shell_safe_json_parse(policy)
+
+    BlobInventoryPolicy, InventoryRuleType, BlobInventoryPolicyName = \
+        cmd.get_models('BlobInventoryPolicy', 'InventoryRuleType', 'BlobInventoryPolicyName')
+    properties = BlobInventoryPolicy()
+    if 'type' not in policy:
+        policy['type'] = InventoryRuleType.INVENTORY
```

Comment:
May I ask why shall cli set a default type for policy? It seems more reasonable for service to set a default value for None properties.

Reply:
at first, for the type property, there is only one value available. secondly, when there is no type in policy schema, service team don't want to break customer, so they would like to have a fallback when there is no type provided, which is also aligning with PSH.